src/main/java/com/iwaconsolti/school/service/SchoolServiceImpl.java

```diff
@@ -0,0 +1,84 @@
+package com.iwaconsolti.school.service;
+
+import com.iwaconsolti.school.model.Course;
+import com.iwaconsolti.school.model.Grade;
+import com.iwaconsolti.school.model.School;
+import com.iwaconsolti.school.model.Student;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.util.List;
+
+public class SchoolServiceImpl implements SchoolService {
+    private final School school;
+
+    private final int limitGrade;
+
+    public SchoolServiceImpl(School school, int limitGrade) {
+        this.school = school;
+        this.limitGrade = limitGrade;
+    }
+
+    @Autowired
+    private CourseService course;
```

Comment:
You should put Service in these cases, since course can be confused with the object Course